crates/ruff_db/src/diagnostic/render/full.rs

```diff
@@ -133,59 +145,81 @@ impl std::fmt::Display for Diff<'_> {
         // tests, which is the only place these are currently used.
         writeln!(f, "ℹ {}", fmt_styled(message, self.stylesheet.separator))?;
 
-        let (largest_old, largest_new) = diff
-            .ops()
-            .last()
-            .map(|op| (op.old_range().start, op.new_range().start))
-            .unwrap_or_default();
+        for range in cells.values() {
```

Comment:
I wonder if we could reduce the overhead for non notebooks by extracing the shared logic for rendering fixes into a separate method and:

* We call that method directly if source isn't a notebook (in which case we can skip all/most of the operations?)
* For notebooks, do whatever mapping is necessary, call that method for each cell